### DISCUSSION TITLE
iptables error and ansible error

### DISCUSSION THREAD

**Discussion by m0n3r0** [ipspace/netlab#1318] [General]
Hi,
First time running netlab, I encounter 2 errors when executing "netlab up"
Any insight how to solve this

WARN[0000] errors during iptables rules install: not available

Error executing ansible-playbook /home/myUnixID/venv-netlab/lib/python3.10/site-packages/netsim/ansible/initial-config.ansible:
  [Errno 2] No such file or directory: 'ansible-playbook'
[FATAL]   Executing Ansible playbook /home/myUnixID/venv-netlab/lib/python3.10/site-packages/netsim/ansible/initial-config.ansible failed
Error executing netlab initial --no-message:
  Command '['netlab', 'initial', '--no-message']' returned non-zero exit status 1.
[FATAL]   netlab up: netlab initial failed, aborting...
(venv-netlab) <user>@HKG-X8031922:~/testlab$

I installed python under venv in my home directory

Thanks,

**Comment by ipspace** [comment]
Second one is easy: you didn't install Ansible.

As for the first one, I'm guessing it's coming from containerlab (here's the matching code: https://github.com/srl-labs/containerlab/blob/91b1fc394f1fea78db9d31cedcac9fc34ba084bd/runtime/docker/docker.go#L353), and digging deeper into the code, it looks like **iptables** is not installed.

Did you use the **netlab install** installation scripts (see https://netlab.tools/install/ubuntu/ and https://netlab.tools/netlab/install/)? If you did, which ones did you start?

**Reply by m0n3r0** [reply]
Thanks!!
I use "python3 -m pip install networklab" and then go straight to install containerlab described in https://netlab.tools/labs/clab/ as it seems the easiest way.
So I skipped this entire page Ubuntu VM Installation --> Manual Virtual Machine Provisioning[¶](https://netlab.tools/install/ubuntu-vm/#manual-virtual-machine-provisioning) as I already have my Ubuntu installed. 
FYI I used Windows Subsytem Linux (WSL) to install Ubuntu

Let me rebuild the venv again.. thanks for insight

**Comment by m0n3r0** [comment]
I still encounter ansible error even I have ansible installed under my python venv.
I decided to install containerlab directly into Ubuntu WSL by clean install WLS Ubuntu from scratch as I see many interfaces created. This is where I discover that containerlab page has notes to install docker under WSL.
I am not sure what is the relation of ansible and docker but I have containerlab running ok without specifically installing ansible

This is copy paste from containerlab notes

Windows Subsystem Linux (WSL)#
Containerlab runs on WSL, but you need to install docker-ce inside the WSL2 linux system instead of using Docker Desktop3.

If you are running Ubuntu/Debian as your WSL2 machine, you can use the quick setup this script to install docker-ce.

PS: I still have iptables error when running containerlab but seems it is not having issues in the lab

**Comment by ipspace** [comment]
The whole toolchain (netlab, Docker, containerlab, Ansible) has to be installed and run on the Ubuntu virtual machine. No idea how to get that done on WSL :( If you figure it out, please post detailed instructions and I'll add them to the documentation.

Also, the netlab installation script should work on the WSL Ubuntu VM.

**Reply by m0n3r0** [reply]
Installation was working but just firing "netlab up" getting error in containerlab case. I will try to mess around next time.
